test/integration/attachment_access_limited_integration_test.rb

```diff
@@ -62,6 +62,27 @@ class AttachmentAccessLimitedIntegrationTest < ActionDispatch::IntegrationTest
       click_button 'Save'
       AssetManagerCreateWhitehallAssetWorker.drain
     end
+
+    it 'marks attachment as access limited in Asset Manager when bulk uploaded to draft document' do
+      visit admin_news_article_path(edition)
+      click_link "Modify attachments"
+      click_link "Bulk upload from Zip file"
+      attach_file 'Zip file', path_to_attachment('sample_attachment.zip')
+      click_button 'Upload zip'
+      fill_in 'Title', with: 'file-title'
+      click_button 'Save'
+
+      Services.asset_manager.expects(:create_whitehall_asset).with do |params|
+        params[:legacy_url_path] =~ /greenpaper\.pdf/ &&
+          params[:access_limited] == ['user-uid']
```

Comment:
Minor: I think you might be able to do something like this here:

```ruby
Services.asset_manager.expects(:create_whitehall_asset).with(has_entries(
  legacy_url_path: regexp_matches(/greenpaper\.pdf/),
  access_limited: ['user-uid'])
)
```

If something like that does work, it might display a more useful error message if the expectation is not satisfied.

Reply:
Thanks @floehopper! I've updated the test to use `has_entries` instead of the custom block matcher. I've left it as two expectations, though, as I think that makes it clearer that we're expecting the PDF and it's thumbnail to be access limited.

Reply:
@chrisroos Cool. I wasn't imagining combining the expectations. What you've done looks right to me!